src/util/cstr/fd_cstr.h

```diff
@@ -53,6 +53,14 @@ FD_FN_PURE  double       fd_cstr_to_double( char const * s );
 
 FD_FN_PURE ulong fd_cstr_to_ulong_octal( char const * s );
 
+/* fd_cstr_to_ip4_addr parses an IPv4 address matching format
+   %u.%u.%u.%u  On success returns the numerical representation of the
+   address in [0;UINT_MAX). On fail returns ULONG_MAX. */
```

Comment:
nit: I wonder if we should typedef IPv4 addresses, with a comment explaining how the address is represented, because they should be interpreted differently to typical `ulong`s.